src/analyzing/AST_to_IL.ml

```diff
@@ -1500,8 +1500,17 @@ and stmt_aux env st =
       python_with_stmt env manager opt_pat body
   (* Java: synchronized (E) S *)
   | G.OtherStmtWithStmt (G.OSWS_Block _, [ G.E objorig ], stmt1) ->
+      (* TODO: Restrict this to a syncrhonized block ? *)
       let ss, _TODO_obj = expr_with_pre_stmts env objorig in
       ss @ stmt env stmt1
+  (* Rust: unsafe block *)
+  | G.OtherStmtWithStmt (G.OSWS_Block ("Unsafe", tok), [], stmt1) ->
+      let ss, _ =
+        with_pre_stmts env (fun env ->
+            let unit = mk_unit (G.fake "()") NoOrig in
+            mk_aux_var ~str:"__TODO_unsafe_block__" env tok unit)
+      in
```

Comment:
This feels like a long way to create the assignment: `__TODO_unsafe_block__(fresh) = unit`.

We're throwing away the result part of `with_pre_stmts`, and we also don't need the "pre" part, because `ss` is used at the beginning of the returned list anyway. (I think there are some other places in this file that use `XXX_with_pre_stmt` when they don't really need to.)

But there doesn't seem to be any existing function to help do this. Just pointing it out, and I don't have really good suggestions. Not sure if creating a helper function to do so would be helpful or would just add more noise.

Reply:
Fair comment, let's add a proper helper for this.